src/microcalibrate/reweight.py

```diff
@@ -11,16 +17,41 @@
 
 
 def reweight(
-    original_weights,
-    loss_matrix,
-    targets_array,
-    dropout_rate=0.1,
-    epochs=2_000,
-    noise_level=10.0,
-    subsample_every=50,
-    learning_rate=1e-3,
-):
+    original_weights: np.ndarray,
+    loss_matrix: pd.DataFrame,
+    targets_array: np.ndarray,
+    dropout_rate: Optional[float] = 0.1,
+    epochs: Optional[int] = 2_000,
+    noise_level: Optional[float] = 10.0,
+    subsample_every: Optional[int] = 50,
+    learning_rate: Optional[float] = 1e-3,
+) -> tuple[np.ndarray, np.ndarray]:
```

Comment:
Can we add a log level default to error: by default I think we should produce only the progress bar